internal/resourcelist/resourcelist.go

```diff
@@ -59,14 +60,24 @@ func FromContainers(containers []corev1.Container) corev1.ResourceList {
 	return res
 }
 
-func AddCoreResources(res corev1.ResourceList, cpu, mem resource.Quantity) {
-	adjustedCPU := res.Cpu()
-	adjustedCPU.Add(cpu)
-	res[corev1.ResourceCPU] = *adjustedCPU
+func AddCoreResources(res, resToAdd corev1.ResourceList) {
+	for resName, resQty := range resToAdd {
+		qty := res[resName]
+		qty.Add(resQty)
+		res[resName] = qty
+	}
+}
 
-	adjustedMemory := res.Memory()
-	adjustedMemory.Add(mem)
-	res[corev1.ResourceMemory] = *adjustedMemory
+func SubCoreResources(res, resToSub corev1.ResourceList) {
+	for resName, resQty := range resToSub {
+		if resQty.Cmp(res[resName]) > 0 {
+			klog.Errorf("Cannot substract %q because it is not found in the current resources", resName)
```

Comment:
let's actually bubble up this error. This will require returning error on `SubCoreResources` and `Deduct`, but it seems the best approach. Swallowing errors (albeit logging like we do here) it's always something to be very wary of.
`Deduct` is different from `Apply` (`Add`) after all, so let's make this evident in the function signature.

Reply:
I agree, will adjust it. thanks for highlighting that!